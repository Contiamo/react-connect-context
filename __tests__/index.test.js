import React from "react"
import renderer from "react-test-renderer"

import { connectContext } from "../"
const Context = React.createContext({ color: "red" })
const Header = ({ children }) => <h1>{children}</h1>
const Content = ({ children, color, myProp }) => (
  <div>
    <p>{children}</p>
    <div>
      I have looked into context and I've seen the color is: {color}
      I also get my own props like {myProp}!
    </div>
  </div>
)

describe("connectContext", () => {
  it("Should map context to props while preserving existing props", () => {
    const ConnectedContent = connectContext(Context.Consumer)(Content)

    class App extends React.Component {
      render() {
        return (
          <div>
            <Header>Welcome to my App!</Header>
            <ConnectedContent myProp="sup">Sup my frand</ConnectedContent>
          </div>
        )
      }
    }
    const tree = renderer
      .create(
        <Context.Provider value={{ color: "red" }}>
          <App />
        </Context.Provider>
      )
      .toJSON()

    expect(tree).toMatchSnapshot()
  })

  it("Should use the custom mergeContextWithProps function if provided", () => {
    const mergeContextWithProps = (context, props) => ({ ...props, ...context })
    const ConnectedContent = connectContext(
      Context.Consumer,
      mergeContextWithProps
    )(Content)
    const AppWithPropsConflict = () => (
      <ConnectedContent myProp="sup" color="red">
        Sup my frand
      </ConnectedContent>
    )

    const tree = renderer
      .create(
        <Context.Provider value={{ color: "green" }}>
          <AppWithPropsConflict />
        </Context.Provider>
      )
      .toJSON()

    expect(tree).toMatchSnapshot()
  })
})
