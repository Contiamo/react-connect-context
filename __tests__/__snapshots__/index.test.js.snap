// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`connectContext Should map context to props while preserving existing props 1`] = `
<div>
  <h1>
    Welcome to my App!
  </h1>
  <div>
    <p>
      Sup my frand
    </p>
    <div>
      I have looked into context and I've seen the color is: 
      red
      I also get my own props like 
      sup
      !
    </div>
  </div>
</div>
`;

exports[`connectContext Should use the custom mergeContextWithProps function if provided 1`] = `
<div>
  <p>
    Sup my frand
  </p>
  <div>
    I have looked into context and I've seen the color is: 
    green
    I also get my own props like 
    sup
    !
  </div>
</div>
`;
